src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            uint32 randomCreature = 0;
+
+            /*
+             * Note: this spell is always cast when there's nothing in the web or there's a foe. Also,
+             * there are 4 serverside spells for this, but we're unable to know which one does what exactly:
+             * 83922 - (Serverside/Non-DB2) Free Webbed Victim, 83923 - (Serverside/Non-DB2) Free Webbed Victim,
+             * 83924 - (Serverside/Non-DB2) Free Webbed Victim, 83925 - (Serverside/Non-DB2) Free Webbed Victim.
+             * It is safe to assume that each summons one of the three foes, and the other summons nothing.
+             * 83927 - Free Webbed Victim summons orcs. We could try finding summonProperties, but it is hard.
+             */
+
+            switch (urand(DATA_RANDOM_0, DATA_RANDOM_3))
+            {
+                case DATA_RANDOM_0:
+                    randomCreature = NPC_BLOODFANG_SCAVENGER;
+                    break;
+                case DATA_RANDOM_1:
+                    randomCreature = NPC_RABID_DOG;
+                    break;
+                case DATA_RANDOM_2:
+                    randomCreature = NPC_GIANT_RABBID_BEAR;
+                    break;
+                case DATA_RANDOM_3:
+                    randomCreature = 0;
+                    break;
+                default:
+                    break;
+            }
+
+            if (randomCreature != 0)
+            {
+                if (Creature* randomAggresiveCreature = caster->SummonCreature(randomCreature, GetHitUnit()->GetPosition()))
+                    randomAggresiveCreature->Attack(caster, true);
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_silverpine_free_webbed_victim_random::HandleHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+enum WebbedOrcSeaDog
+{
+    SPELL_SINISTER_STRIKE                       = 60195,
+
+    EVENT_WEBBEB_ORC_CHECK_PLAYER               = 1,
+    EVENT_WEBBEB_ORC_TALK                       = 2,
+    EVENT_SINISTER_STRIKE                       = 3,
+
+    TALK_WEBBEB_ORC_FREED                       = 0
+};
+
+// 44942 - Orc Sea Dog
+struct npc_silverpine_orc_sea_dog : public ScriptedAI
+{
+    npc_silverpine_orc_sea_dog(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        me->SetReactState(REACT_ASSIST);
+    }
+
+    void JustAppeared() override
+    {
+        me->SetPowerType(POWER_ENERGY);
+        me->SetMaxPower(POWER_ENERGY, 100);
+        me->SetPower(POWER_ENERGY, 100, true);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+        {
+            _playerGUID = player->GetGUID();
+
+            player->KilledMonsterCredit(NPC_ORC_SEA_DOG, _playerGUID);
```

Comment:
`_playerGUID` is unneeded here. In `KilledMonsterCredit` the guid is used to get a creature, which is never true for players

Reply:
Feedback applied.